lib/src/main/java/com/auth0/jwt/JWTCreator.java

```diff
@@ -59,6 +63,14 @@ private JWTCreator(Algorithm algorithm, Map<String, Object> headerClaims, Map<St
         private final Map<String, Object> payloadClaims;
         private Map<String, Object> headerClaims;
 
+        private static final Set<Class<?>> SUPPORTED_TYPES = new HashSet<>();
+        private static final Set<Class<?>> SUPPORTED_ARRAY_TYPES = new HashSet<>();
+        static {
+            SUPPORTED_ARRAY_TYPES.addAll(Arrays.asList(Integer[].class, Long[].class, String[].class));
+            SUPPORTED_TYPES.addAll(
+                    Arrays.asList(String.class, Integer.class, Double.class, Long.class, Date.class, Boolean.class));
```

Comment:
To support objects of objects in claims, I suggest to add here Map.class.

Then below, in the isValidClaimsValue, you'd need to modify the check as

```
	    for(Class typ: SUPPORTED_TYPES) {
		if( typ.isAssignableFrom(clazz) ) return true;
	    }
            return false;
```

That works, I have tested it for both generation and decoding.

Reply:
You are right, it works with Object of Objects if we add Map.class, but it won't be that one only. If we are supporting Map.class, then we MAY need to support Collections as well. I'm not sure second level objects of non-primitive will be allowed in this lib.

I agree that it will greatly help to construt complex/custom JWT's with that support. For now, I try to keep this PR as it is unless someone from library second this addition. Thank you.

Reply:
Sure. For the time being, I'm using a modified lib built from your branch because I need to deal with nested objects coming from outside authorization service - today. When your code is merged, I'll try and push for nested objects, with a separate merge request.